File: 075 Sort Colors/sort_colors.cs
public class Solution {	// Author: Xinyu Chen
    public void SortColors(int[] nums)
    {
        int pos = 0, red = 0, blue = nums.Length - 1;
        while (pos <= blue)
        {
            if (nums[pos] == 0)
            {
                SwapInt(ref nums[pos++], ref nums[red++]);
            }
            else if (nums[pos] == 2)
            {
                SwapInt(ref nums[pos], ref nums[blue--]);
            }
            else pos++;
        }
    }

    private void SwapInt(ref int a, ref int b)
    {
        int c = a;
        a = b;
        b = c;
    }
}

File: 075 Sort Colors/sort_colors.c
void sortColors(int* nums, int numsSize) {
    int red = 0, blue = numsSize - 1, curr = 0;
    while (curr <= blue) {
        if (nums[curr] == 0) {
            nums[curr++] = nums[red];
            nums[red++] = 0;
        } else if (nums[curr] == 2) {
            nums[curr] = nums[blue];
            nums[blue--] = 2;
        } else {
            curr++;
        }
    }
} /* Author: XC */
